### ISSUE TITLE
NX Daemon failed to start the daemon process

### ISSUE THREAD

**Issue by Roozenboom** [nrwl/nx#16809]
### Current Behavior

In our NX workspace, the NX Daemon process is not starting up or connecting in time. Because of that the NX Daemon is not able of computing the project graph.
```
>  NX   Nx Daemon was not able to compute the project graph.

   Log file with the error: /node_modules/.cache/nx/d/daemon-error.log
   Please file an issue at https://github.com/nrwl/nx
   Nx Daemon is going to be disabled until you run "nx reset".
``` 
and in the end, commands are failing with an error message from the daemon:
```
 >  NX   Cannot read properties of null (reading 'sendMessage')
```

The reason for this error is that our NX workspace is quite large (+- 3500 nx projects and 150.000 files) and it takes the native hasher about 40 seconds to hash all these files. When the Daemon server starts it polls for 10 seconds (packages/nx/src/daemon/client/client.ts L347-363) if the server is running, but in our scenario that is never the case. When I monkey patch the polling to a minute the daemon server starts and the client connects to it as I would expect. 

### Expected Behavior

When the NX Daemon is not running it is started in the background and the client waits long enough for the server to start up so that it could be used for the project graph and caching/hashing. 

### Steps to Reproduce

Prerequisite is that you have a workspace with a lot of files so that the hasher takes more than 10 seconds 
1. nx reset
2. nx run lint <project> (or any other nx command)
  - the 'sendMessageToDaemon' method will trigger 'startInBackground' if the daemon is not available
  - startInBackground will spawn the server in a child process and poll for 10 seconds if the server started correctly
  - after 10 seconds it will throw an exception

### Nx Report

```shell
nx                      : 15.8.8
   @nrwl/js                : 15.8.8
   @nrwl/jest              : 15.8.8
   @nrwl/linter            : 15.8.8
   @nrwl/workspace         : 15.8.8
   @nrwl/angular           : 15.8.8
   @nrwl/cli               : 15.8.8
   @nrwl/cypress           : 15.8.8
   @nrwl/devkit            : 15.8.8
   @nrwl/eslint-plugin-nx  : 15.8.8
   @nrwl/node              : 15.8.8
   @nrwl/react             : 15.8.8
   @nrwl/rollup            : 15.8.8
   @nrwl/tao               : 15.8.8
   @nrwl/web               : 15.8.8
   @nrwl/webpack           : 15.8.8
   typescript              : 4.9.4
   ---------------------------------------
   Community plugins:
   rxjs                        : 6.6.7
   @ionic/angular              : 6.3.0
   @ionic/angular-toolkit      : 6.1.0
   @ngneat/spectator           : 14.0.0
   @nguniversal/builders       : 15.2.0
   @nguniversal/express-engine : 15.2.0
```

### Failure Logs

```shell
I added some additional logging to the server start command

[NX Daemon Server] - 2023-05-05T11:07:12.437Z - Start NX Daemon server [packages/nx/src/daemon/server/server.ts L352]
[NX Daemon Server] - 2023-05-05T11:07:12.457Z - await defaultFileHasher.ensureInitialized(); start [packages/nx/src/daemon/server/server.ts L361]
[NX Daemon Server] - 2023-05-05T11:07:58.939Z - await defaultFileHasher.ensureInitialized(); end [packages/nx/src/daemon/server/server.ts L361]
[NX Daemon Server] - 2023-05-05T11:07:58.940Z - Started listening on: /var/folders/kn/89k9jlnn7c30ck7pbc89zpqc0000gn/T/a8d0b2cd79eda04fc586/d.sock
[NX Daemon Server] - 2023-05-05T11:07:58.946Z - [WATCHER]: Subscribed to changes within: *
[NX Daemon Server] - 2023-05-05T11:08:08.505Z - Established a connection. Number of open connections: 1
[NX Daemon Server] - 2023-05-05T11:08:08.507Z - Closed a connection. Number of open connections: 0
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

I am happy to provide a fix for this issue by increasing the interval time in the `packages/nx/src/daemon/client/client.ts`.

**Comment by bintoll** [comment]
Hello! I have exactly the same issue while building app inside of the docker container, I have much less projects then topic starter. And I confirm that the hot fix that @Roozenboom provided works. Thanks! 
Probably interval that checks for deamon being running should be upgraded so developer would be able to configure timing and amount of attempts

**Comment by sedghi** [comment]
same issue in with building app inside docker container here too. For some reason this does not happen in windows for my colleague

**Comment by mklueh** [comment]
For me it happens in Windows 11, but not in Ubuntu WSL

**Comment by premier213** [comment]
for me is fixed by downgrade to `16.4.1`
